api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -34,6 +34,7 @@ type HostedControlPlane struct {
 }
 
 // HostedControlPlaneSpec defines the desired state of HostedControlPlane
+// +kubebuilder:validation:XValidation:rule=`self.capabilities == oldSelf.capabilities`, message="Capabilities is immutable. Changes might result in unpredictable and disruptive behavior."
```

Comment:
can we instead have this validation at the field level with self == oldself
e.g. like with IssuerURL

Reply:
I'll try that, thanks!